packages/core/src/BuildContractVersionHashes.js

```diff
@@ -0,0 +1,127 @@
+const assert = require("assert");
+const path = require("path");
+const fs = require("fs");
+
+const { toWei, utf8ToHex, padRight, soliditySha3 } = web3.utils;
+
+const { createConstructorParamsForContractVersion, interfaceName } = require("@uma/common");
+const { getTruffleContract } = require("../index");
+
+async function buildHashes(contractType) {
+  assert(contractType == "Perpetual" || contractType == "ExpiringMultiParty", "Invalid contract type defined!");
+
+  const contractCreator = (await web3.eth.getAccounts())[0];
+
+  const FinancialContract = getTruffleContract(contractType, web3);
+  const Finder = getTruffleContract("Finder", web3);
+  const IdentifierWhitelist = getTruffleContract("IdentifierWhitelist", web3);
+  const AddressWhitelist = getTruffleContract("AddressWhitelist", web3);
+  const MockOracle = getTruffleContract("MockOracle", web3);
+  const Token = getTruffleContract("ExpandedERC20", web3);
+  const SyntheticToken = getTruffleContract("SyntheticToken", web3);
+  const Timer = getTruffleContract("Timer", web3);
+  const Store = getTruffleContract("Store", web3);
+  const ConfigStore = getTruffleContract("ConfigStore", web3);
+  const OptimisticOracle = getTruffleContract("OptimisticOracle", web3);
+
+  const identifier = "TEST_IDENTIFIER";
+  const fundingRateIdentifier = "TEST_FUNDiNG_IDENTIFIER";
+
+  const finder = await Finder.new({ from: contractCreator });
```

Comment:
I tried this! I wish it was this easy :(

the byetcodes don't match :/

Reply:
Just to humor me, could you try doing it this way?
```
const { deployedBytecode } = getTruffleContract("Voting", web3);
```

I think that will return a different result than:
```
const { deployedBytecode } = getArtifact("Voting");
```

In the test env, I think the latter will return the truffle-compiled deployedBytecode, but the former will return the hardhat deployedBytecode because of some of the weirdness in how we deal with hardhat's test env.

Maybe to put it differently, assuming that both of those return different results, do _either_ of those work?

Reply:
In non-test environments they match. `getArtifact` is actually not exported from `core` and `getTruffleContract` calls `getArtifact`. I modified the `getArtifact to log a hash of the bytecode hash that comes back and it's the same as that produced between `getArtifact` and `getTruffleContract`.

In a test environment, you are right that there is a mismatch between these hashes. but I don't think it matters as we use use the `getTruffleContract` everywhere and so the hash produced by this build process maps to the hash used in the tests.